client/platforms/windows/daemon/windowssplittunnel.cpp

```diff
@@ -273,58 +443,60 @@ std::vector<uint8_t> WindowsSplitTunnel::generateAppConfiguration(
 
   auto config = reinterpret_cast<IP_ADDRESSES_CONFIG*>(&out[0]);
 
   auto ifaces = QNetworkInterface::allInterfaces();
 
-  if (vpnAdapterIndex == 0) {
+    if (vpnAdapterIndex == 0) {
     vpnAdapterIndex = WindowsCommons::VPNAdapterIndex();
   }
-
   // Always the VPN
-  getAddress(vpnAdapterIndex, &config->TunnelIpv4,
-             &config->TunnelIpv6);
-  // 2nd best route
-  getAddress(inetAdapterIndex, &config->InternetIpv4, &config->InternetIpv6);
+  if (!getAddress(vpnAdapterIndex, &config->TunnelIpv4,
+                  &config->TunnelIpv6)) {
+    return {};
+  }
+  // 2nd best route is usually the internet adapter
+  if (!getAddress(inetAdapterIndex, &config->InternetIpv4,
+                  &config->InternetIpv6)) {
+    return {};
+  };
   return out;
 }
-void WindowsSplitTunnel::getAddress(int adapterIndex, IN_ADDR* out_ipv4,
+bool WindowsSplitTunnel::getAddress(int adapterIndex, IN_ADDR* out_ipv4,
                                     IN6_ADDR* out_ipv6) {
   QNetworkInterface target =
       QNetworkInterface::interfaceFromIndex(adapterIndex);
   logger.debug() << "Getting adapter info for:" << target.humanReadableName();
 
-  // take the first v4/v6 Adress and convert to in_addr
-  for (auto address : target.addressEntries()) {
-    if (address.ip().protocol() == QAbstractSocket::IPv4Protocol) {
-      auto adrr = address.ip().toString();
-      std::wstring wstr = adrr.toStdWString();
-      logger.debug() << "IpV4" << logger.sensitive(adrr);
-      PCWSTR w_str_ip = wstr.c_str();
-      auto ok = InetPtonW(AF_INET, w_str_ip, out_ipv4);
-      if (ok != 1) {
-        logger.debug() << "Ipv4 Conversation error" << WSAGetLastError();
+  auto get = [&target](QAbstractSocket::NetworkLayerProtocol protocol) {
+    for (auto address : target.addressEntries()) {
+      if (address.ip().protocol() != protocol) {
+        continue;
       }
-      break;
+      return address.ip().toString().toStdWString();
     }
+    return std::wstring{};
+  };
+  auto ipv4 = get(QAbstractSocket::IPv4Protocol);
+  auto ipv6 = get(QAbstractSocket::IPv6Protocol);
+
+  if (InetPtonW(AF_INET, ipv4.c_str(), out_ipv4) != 1) {
+    logger.debug() << "Ipv4 Conversation error" << WSAGetLastError();
+    return false;
   }
-  for (auto address : target.addressEntries()) {
-    if (address.ip().protocol() == QAbstractSocket::IPv6Protocol) {
-      auto adrr = address.ip().toString();
-      std::wstring wstr = adrr.toStdWString();
-      logger.debug() << "IpV6" << logger.sensitive(adrr);
-      PCWSTR w_str_ip = wstr.c_str();
-      auto ok = InetPtonW(AF_INET6, w_str_ip, out_ipv6);
-      if (ok != 1) {
-        logger.error() << "Ipv6 Conversation error" << WSAGetLastError();
-      }
-      break;
-    }
+  if (ipv6.empty()) {
+    std::memset(out_ipv6, 0x00, sizeof(IN6_ADDR));
+    return true;
+  }
+  if (InetPtonW(AF_INET6, ipv6.c_str(), out_ipv6) != 1) {
+    logger.debug() << "Ipv6 Conversation error" << WSAGetLastError();
+    return false;
```

Comment:
Not sure about this part. It breaks app split tunnel when ipv6 is not setup. 

We can just ignore then ipv6 address is not setup.